### ISSUE TITLE
setUserId

### ISSUE THREAD

**Issue by htaoufik** [piwik/piwik-dotnet-tracker#17]
Hello, In setUserId function of PiwikTracker:

How can we set the userId if 

1-  By default it is null ( this.userId = null; in constructor )
2- Before changing the userId value we check if the current userId is null or empty and throw an exception when it is the case.

Got this in decompiled code from nuget PiwikTracker v 2.7

public void setUserId(string userId)
    {
      if (userId == null)
      {
        this.setNewVisitorId();
      }
      else
      {
        if (string.IsNullOrEmpty(this.userId))
          throw new Exception("User ID cannot be empty.");
        this.userId = userId;
      }
    }

I think we shall replace  if (string.IsNullOrEmpty(this.userId)) by  if (string.IsNullOrEmpty(userId))

Sincerely.
H.T

**Comment by JulienMoumne** [comment]
You have found a bug thank you.

The bug is at line https://github.com/piwik/piwik-dotnet-tracker/blob/2.7.0/Piwik.Tracker/PiwikTracker.cs#L1103

```C#
            if (string.IsNullOrEmpty(this.userId)) {
                throw new Exception("User ID cannot be empty.");
            }
```

should be 

```C#
            if (string.IsNullOrEmpty(userId)) {
                throw new Exception("User ID cannot be empty.");
            }
```

As in https://github.com/piwik/piwik-php-tracker/blob/1.0.1/PiwikTracker.php#L1119

```PHP
        if ($userId === '') {
            throw new Exception("User ID cannot be empty.");
        }
```

**Comment by JulienMoumne** [comment]
Thanks for reporting the bug.
It has been fixed and published in version 2.8.0 : https://www.nuget.org/packages/Piwik.Tracker/2.8.0
